.vscode/settings.json

```diff
@@ -2,6 +2,7 @@
     "python.formatting.provider": "black",
     "[python]": {
         "editor.formatOnSave": true,
+        "editor.defaultFormatter": "ms-python.black-formatter",
```

Comment:
I'm surprised that the `"python.formatting.provider"` option above wasn't enough. Didn't autoformatting work for you?

Reply:
I don't know. Sometimes VSCode has its own agenda and adds things to the settings without asking...